kuma/wiki/apps.py

```diff
@@ -94,11 +94,17 @@ def on_document_save(self, sender, instance, **kwargs):
         - trigger the cache invalidation of the contributor bar for the given
           document
         """
+        # cleanup document zones
         async = kwargs.get('async', True)
         invalidate_zone_urls_cache(instance, async=async)
         invalidate_zone_stack_cache(instance, async=async)
+
+        # refresh the document's contributor bar data
         DocumentContributorsJob().invalidate(instance.pk)
 
+        # new generation of document's code samples
+        DocumentCodeSampleJob(generation_args=[instance.pk]).renew()
```

Comment:
This should be added to the method docstring, or the docstring should be less wordy. Also, are you worried about creating a long-lived generation key for every page, even ones without code samples?